stackit/internal/features/beta.go

```diff
@@ -0,0 +1,39 @@
+package features
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/core"
+)
+
+// BetaFeatureFlagEnabled returns whether this provider is running with the BETA feature flag (enable_beta) enabled.
+func BetaFeatureFlagEnabled(data core.ProviderData) bool {
+	return data.EnableBeta
+}
+
+// BetaResourcesEnabled returns whether this provider has BETA functionality enabled.
+//
+// In order of precedence, beta functionality can be managed by:
+//   - Environment Variable `STACKIT_TERRAFORM_BETA_RESOURCES` - `true` is enabled, unset or any other value is disabled.
+//   - Provider configuration feature flag `enable_beta` - `true` is enabled, `false` is disabled.
+func BetaResourcesEnabled(data core.ProviderData) bool {
+	value, ok := os.LookupEnv("STACKIT_TERRAFORM_BETA_RESOURCES")
```

Comment:
And would maybe rename this to match the provider field
```suggestion
	value, ok := os.LookupEnv("STACKIT_TF_ENABLE_BETA_RESOURCES")
```